app/models/user.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  has_many :posts, dependent: :delete_all
+  has_many :likes
+  has_many :comments
+  has_many :incoming_requests, foreign_key: 'requested_user_id', class_name: 'Friendship'
+  has_many :outcoming_requests, foreign_key: 'requesting_user_id', class_name: 'Friendship'
+  devise :omniauthable, omniauth_providers: %i[facebook]
+
+  def self.from_omniauth(auth)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+      user.email = auth.info.email
+      user.password = Devise.friendly_token[0, 20]
+      user.name = auth.info.name # assuming the user model has a name
+      user.image = auth.info.image # assuming the user model has an image
```

Comment:
- [ ] Also the user doesn't have an image. I would suggest that you add `name` to your user model. But if you application doesn't fully depend on it, you can remove those lines.